includes/classes/DistributorPost.php

```diff
@@ -0,0 +1,563 @@
+	 * for live updates of the original post URL for internal connections.
+	 *
+	 * @var string
+	 */
+	private $original_post_url = '';
+
+	/**
+	 * The type of connection this post is distributed from.
+	 *
+	 * @var string internal|external|pushed|empty (for source)
+	 */
+	public $connection_type = '';
+
+	/**
+	 * The connection ID this post is distributed from.
+	 *
+	 * For internal connections this is the site ID. For external connections
+	 * this refers to the connection ID.
+	 *
+	 * @var int
+	 */
+	public $connection_id = 0;
+
+	/**
+	 * The source site data for internal connections.
+	 *
+	 * This is an array of site data for the source site. This is set by
+	 * the populate_source_site() method upon access to avoid switching
+	 * sites unnecessarily.
+	 *
+	 * @var array {
+	 *    @type string $home_url The site's home page.
+	 *    @type string $site_url The site's WordPress address.
+	 *    @type string $rest_url The site's REST API address.
+	 *    @type string $name     The site name.
+	 * }
+	 */
+	private $source_site = [];
+
+	/**
+	 * Initialize the DistributorPost object.
+	 *
+	 * @param WP_Post|int $post WordPress post object or post ID.
+	 */
+	public function __construct( $post ) {
+		$post = get_post( $post );
+
+		if ( ! $post ) {
+			return;
+		}
+
+		$this->post = $post;
+
+		// Set up the distributable data.
+		$this->meta  = Utils\prepare_meta( $post->ID );
+		$this->terms = Utils\prepare_taxonomy_terms( $post->ID );
+		$this->media = $this->prepare_media();
+
+		/*
+		 * The original post ID is listed as excluded post meta and therefore
+		 * unavailable in the meta property. We need to get it using the post
+		 * meta API.
+		 */
+		$original_post_id = get_post_meta( $post->ID, 'dt_original_post_id', true );
+		if ( empty( $original_post_id ) ) {
+			// This is the source post.
+			$this->is_source         = true;
+			$this->is_linked         = true;
+			$this->original_post_id  = $this->post->ID;
+			$this->original_post_url = get_permalink( $this->post->ID );
+			return;
+		}
+
+		// Set up information for a distributed post.
+		$this->is_source = false;
+		// Reverse value of the `dt_unlinked` meta data.
+		$this->is_linked         = ! get_post_meta( $post->ID, 'dt_unlinked', true );
+		$this->original_post_id  = $original_post_id;
+		$this->original_post_url = get_post_meta( $post->ID, 'dt_original_post_url', true );
+
+		// Determine the connection type.
+		if ( get_post_meta( $post->ID, 'dt_original_blog_id', true ) ) {
+			/*
+			 * Internal connections store the original blog's ID.
+			 *
+			 * Pushed and pulled posts are indistinguishable from each other.
+			 */
+			$this->connection_type = 'internal';
+			$this->connection_id   = get_post_meta( $post->ID, 'dt_original_blog_id', true );
+		} elseif ( get_post_meta( $post->ID, 'dt_full_connection', true ) ) {
+			// This connection was pushed from an external connection.
+			$this->connection_type = 'pushed';
```

Comment:
Wondering if we need another property to store if something is pushed or pulled? I guess just seems a little weird to me to have a `$connection_type` property but instead of storing `external` there, we store `pushed`.

I understand there's not a way to distinguish pushed or pulled on internal content, so in essence this is `external - pushed` and the other is `external - pulled` but might be more obvious to have a different property to store if it's a push or pull.

Reply:
See 768515f101c01b6c0e32534b4af2528a38548f40